CollAction/Controllers/ProjectsController.cs

```diff
@@ -66,14 +67,24 @@ public ViewResult StartInfo()
         }
 
         // GET: Projects/Details/5
-        public async Task<IActionResult> Details(int? id)
+        public async Task<IActionResult> DetailsById(int? id)
         {
-            if (id == null)
+            var project =  await _service.GetProjectById(id); 
+            if (project == null)
             {
                 return NotFound();
             }
-
-            List<DisplayProjectViewModel> items = await DisplayProjectViewModel.GetViewModelsWhere(_context, p => p.Id == id && p.Status != ProjectStatus.Hidden && p.Status != ProjectStatus.Deleted);
+            string validUrlForProjectName = WebUtility.UrlEncode(project.Name);
+            if(String.IsNullOrEmpty(validUrlForProjectName))
+            {
+                return NotFound();
+            }
+            return LocalRedirect("~/projects/"+validUrlForProjectName+"/details");
+        }
+        
+        public async Task<IActionResult> Details(string name)
+        {
+            List<DisplayProjectViewModel> items = await DisplayProjectViewModel.GetViewModelsWhere(_context, p => p.Name == WebUtility.UrlDecode(name) && p.Status != ProjectStatus.Hidden && p.Status != ProjectStatus.Deleted);
```

Comment:
Things should be decoded automatically. I don't think WebUtility.UrlDecode is necessary here (maybe test that, not 100% sure).

Reply:
Apparently due to some reason this is not always the case ( checked it ), so it's better to decode it always. Could have something to do with redirect.

Reply:
Do you know which circumstances this happens? Because double decoding might give rise to other issues.

Suppose a project name has something in it that happens to be a valid url encoding, then double decoding it will mean we won't find the project because the name'll be wrong.

If it's that unreliable, maybe we should go with the id + name idea. That way we don't need to mess with the name that much.